fdbserver/TLogServer.actor.cpp

```diff
@@ -3425,11 +3471,32 @@ ACTOR Future<Void> restorePersistentState(TLogData* self,
 					// logData->version.get());
 
 					if (logData) {
+						if (SERVER_KNOBS->ENABLE_VERSION_VECTOR_TLOG_UNICAST && spillTargetLogDataId.present() &&
+						    qe.id == spillTargetLogDataId.get() && qe.version < logData->persistentDataDurableVersion) {
+							logData->unknownCommittedVersions.emplace_front(qe.version, qe.prevVersion, qe.tLogLocIds);
+
+							while (!logData->unknownCommittedVersions.empty() &&
+							       logData->unknownCommittedVersions.back().version <= logData->knownCommittedVersion) {
+								logData->unknownCommittedVersions.pop_back();
+							}
```

Comment:
Can you make a function for these lines for code reuse?
There are similar code below and in `tLogCommit()`.

Reply:
Good point. Done now.